### DISCUSSION TITLE
How to get binaries with sbin, and also man pages, and completions?

### DISCUSSION THREAD

**Discussion by ngocphamm** [zdharma-continuum/zinit#439] [Q&A] [answered]
I'm trying to replace a few binaries from being managed by Homebrew to `zinit`. So far this works well that the binaries are inside `$ZPFX/polaris`. However, I noticed that they do not have completions automanaged, as well as man pages.

```
zinit as"null" wait"0a" lucid light-mode from"gh-r" sbin for \
  achannarasappa/ticker \
  tarkah/tickrs \
  jesseduffield/lazydocker \
  jesseduffield/lazygit \
  ekzhang/bore \
  sbin"**/rg" BurntSushi/ripgrep \
  sbin"**/fd" @sharkdp/fd \
  sbin"**/bat" @sharkdp/bat \
  sbin"**/delta" dandavison/delta \
  sbin"bin/exa" ogham/exa
```

After a while, I understand that the `as"null"` ice is the culprite, as it basically is

```
# as"null" – a shorthand for pick"/dev/null" nocompletions
# i.e.: it disables the default script-file sourcing and also the installation of completions.
```

I thought `completions` ice can be use to overwrite `as"null"`, according to https://github.com/zdharma-continuum/zinit#completions, but when I write the above like `... completions sbin"bin/exa: ogham/exa`, `zinit` will try to download `completions` like it thinks it's a program or something.

Another weird thing is when I enter `zi creinstall ` and hit TAB, I can see it autocomplete with a list of plugins (I assume) available for installing completions. I tried with `BurntSushi/ripgrep` and yes it installs completions for that and I can type `rg --TAB` from that point on. However, it doesn't show `ogham/exa` in that list. Does that mean it doesn't recognize that `exa` has completions for it to try installing? I think exa does have completions, though.

![CleanShot 2022-12-09 at 18 27 34@2x](https://user-images.githubusercontent.com/395123/206811509-86af92bd-b0a6-43dc-9ab5-a6e85aa639f2.png)

So for now, I'm stuck, and don't know yet what to use.

Anyone has gone this direction and have some ideas?

**Comment by ngocphamm** [comment]
Turns out I didn't have latest version of `zinit` which has `completions` ICE added a few days ago.

Now I can have the for-syntax like `zinit as"null" wait"0a" lucid light-mode from"gh-r" sbin completions for \` and it will install completions for plugins if available. 🎉 

Now I wish I could do something similar for the man files 🙏  (or maybe it already exists and I just haven't found it yet).

**Comment by vladdoster** [comment, marked as answer]
@ngocphamm For manpages, install the [zinit-annex-linkman](https://github.com/zdharma-continuum/zinit-annex-linkman) annex.

For example:

```zsh
zi as'completions' from'gh-r' id-as lbin'!' lman light-mode null for \
  @sharkdp/{fd,hyperfine}
```

<img width="2056" alt="Screenshot 2022-12-22 at 19 52 17" src="https://user-images.githubusercontent.com/10052309/209255147-d80796fc-8ffb-4da6-ba1e-6e38d42d8d5a.png">

**Reply by ngocphamm** [reply]
Oh thanks @vladdoster. This is perfect! Got it working well!

**Comment by alextorma** [comment]
@vladdoster this doesn't seem to be working anymore. I have the following in my zshrc:
```zsh
zi as'completions' from'gh-r' id-as lbin'!' lman light-mode null for \
    @sharkdp/{fd,bat} 
```

The binary and man pages install, but the completions don't seem to load. Any suggestions?
